### PR TITLE
Add link to latest mac build and move to build folder (solves #1926)

### PR DESCRIPTION
**Pull request by erf**
Add link to latest mac build and move to build folder. Solves #1926

### CODE REVIEW

File: ci/deploy.sh [hunk 1]
```diff
@@ -96,6 +95,12 @@ case $OSTYPE in
     SHA256=$(shasum -a 256 $zipname | cut -d' ' -f1)
     sed -e "s/@TAG@/$TAG_NAME/g" -e "s/@SHA256@/$SHA256/g" < ci/wezterm-homebrew-macos.rb.template > wezterm.rb
 
+    latest=WezTerm-macos-latest.zip
+    ln -sf $zipname $latest
+    mkdir -p build
+    (cd build ; rm -rf $zipdir $zipname $latest)
+    mv $zipname $zipdir $latest build
```
Comment (wez) [inline ci/deploy.sh:102]: This part will break the nightly and release uploads:

```
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: WezTerm-*.zip. No artifacts will be uploaded.
```

Putting the artifacts in subdirs is problematic with the way that artifact uploading works with GH, so I don't want to do this bit.

Comment (erf) [reply ci/deploy.sh:102]: Ok. I can remove the last commit and just leave the link in the main folder.

Comment (erf) [reply ci/deploy.sh]: Done.

Comment (wez) [reply ci/deploy.sh]: Thanks!  Now, the other problem is that this is uploading two copies of the artifacts!

```
Run actions/upload-artifact@v3
With the provided path, there will be 2 files uploaded
```

I think the sanest way to deal with this is to add either command line parameter(s) or an environment variable that allows you to tell the script whether to use the naming scheme used for distribution, or whether to use a convenient name for local testing, or perhaps, just let you completely override the name.

Actually, I think this already exists; 

https://github.com/wez/wezterm/blob/efcc4b48c11fa65d17a6fa9d050b0ac2eef4f48a/ci/deploy.sh#L22-L27

Can you try running it like this:

```
TAG_NAME=latest ./ci/deploy.sh
```

and see if that has the result you're looking for?
I think that will output `WezTerm-macos-latest.zip`

Comment (erf) [reply ci/deploy.sh]: Ah, you are right - that solves it 😅👍

**Pull request closed without merging**